### PR TITLE
Added gh-actions YAML file and config

### PR DESCRIPTION
**Pull request by GhostOps77**
Currently these tests gonna fail, that's because of mypy
but still waiting to merge all those changes

Solves #74 and #80

### CODE REVIEW

File: bin/testrepl.py [hunk 1]
```diff
@@ -1,25 +0,0 @@
-import click
-from click_repl import register_repl
-
```
Comment (auvipy) [inline bin/testrepl.py:3]: why should we remove this file?

**Review by auvipy — CHANGES_REQUESTED** [review]

File: bin/testrepl.py (hunk 1 above)
Comment (GhostOps77) [reply bin/testrepl.py:3]: that test case is not at all being involved
I thought of adding that test in the next pr

If u want me to add that test, I can make an another commit for that

**Comment by GhostOps77** [comment]
How about now?

File: tests/test_repl.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    @cli.command()
+    @click.option("--foo", is_flag=True)
+    def bar(foo):
+        print("Bar!")
+
+
+    register_repl(cli)
+
+    with pytest.raises(SystemExit):
+        cli()
```
Comment (auvipy) [inline tests/test_repl.py:27]:
```suggestion
        cli()

```

**Review by auvipy — APPROVED** [review]

**Comment by GhostOps77** [comment]
@auvipy  Next I'm gonna remove flake8 and add black formatter instead
Is that ok for you?

**Comment by auvipy** [comment]
ok

**Comment by auvipy** [comment]
and rebase the big pr

**Pull request merged**